src/HtmlFramework/Widget/BlockOFun.php

```diff
@@ -0,0 +1,67 @@
+<?php declare(strict_types=1);
+
+namespace HtmlFramework\Widget;
+
+use Utils\HtmlUtils;
+use HtmlFramework\Packet\ArticlePacket;
+use HtmlFramework\Widget\WidgetTrait;
+
+class BlockOFun {
+   use WidgetTrait;
+
+   const FUN_IMAGE_WIDTH = 140;
+   const FUN_IMAGE_HEIGHT = 140;
+   const FUN_CLASS = 'block-o-fun';
+
+   public static function getHtmlFromArticlePacket(ArticlePacket $aPacket): string {
+      return (new self())->getHtml();
+   }
+
+   public function getHtml(): string {
+      $funElements = [];
+      $funElements[] = HtmlUtils::makeH1Element('This is your basic block of Fun!', 'fun');
+      $funElements[] = $this->makeImageElement();
+      for ($i = 1; $i <= 4; $i++) {
+         $funElements[] = HtmlUtils::addRandomFun($this->getText($i), rand(0, 100));
+         $funElements[] = HtmlUtils::makePageWhitespace();
+      }
+      $fpHtml = HtmlUtils::makePElement(implode(' ', $funElements));
+      return HtmlUtils::makeDivElement($fpHtml, ['class' => 'block-o-fun']);
+   }
+
+   protected function renderWidget(): bool {
```

Comment:
Should move into base?
 * I think in general we'll want it rendered with the option to turn off

Reply:
[a98b872](https://github.com/cdcline/demo-website/pull/55/commits/a98b872efb81b96f586ab6eb80799b2f3cacd326)